File: exchange/webexchange/static/wallet_manage/wallet_manage.js
window.onload = () => {
    const csrftoken = getCookie('csrftoken');
    const username = $("#user-name").html();
    let data = {
        username: username,
    }
    let params = {
        method: "POST",
        headers: {
            'Content-Type': 'application/json',
            'X-CSRFToken': csrftoken,
        },
    }
    
    data.register = 'no'
    params.body = JSON.stringify(data);
    let wallet_list = document.getElementById('wallet-table');
    fetch(`/wallet/${username}/`, params)
    .then(response => response.json())
    .then(data => {
        if (data.wallets_data == undefined) {
            // no wallets
            let tr = document.createElement('tr');
            let td = document.createElement('td');
            tr.id = 'no-register-tr';
            td.id = 'no-register-td';
            td.innerHTML = "No Registered Wallet";
            tr.appendChild(td);
            wallet_list.appendChild(tr);
        } else {
            // show wallets 
            for (item_data of data.wallets_data) {
                tr = document.createElement('tr');
                td = document.createElement('td');
                td.innerHTML = item_data.wallet_ID;
                tr.appendChild(td);
                wallet_list.appendChild(tr);
            }
        }
    })

    $("#add-button").click(() => {
        // register wallet
        data.register = 'yes';
        data.wallet_ID = $("#wallet-id").val()
        params.body = JSON.stringify(data);
        
        fetch(`/wallet/${username}/`, params)
        .then(response => response.json())
        .then(data => {
            // add wallet data success
            if (data.register == 'success') {
                alert(`Register wallet ${data.wallet_id} success!`);
                tr = document.createElement('tr');
                td = document.createElement('td');
                td.innerHTML = data.wallet_id;
                tr.appendChild(td);
                wallet_list.appendChild(tr);
                $("#wallet-id").val("");
            }
            else {
                alert("Register Failed!");
            }
        })
        .catch(error => console.log(error));
    })
}

File: exchange/webexchange/views/wallet_manage.py
from webexchange.views.common.utils import *

class wallet(View):
    def get(self, request, **kwargs):
        username = kwargs.get('username')
        if username != "":
            return render(request, 'wallet.html', context={'username': username})
        else:
            return redirect('404')
        
    def post(self, request, **kwargs):
        response = {'alert': None}
        data = json.loads(request.body.decode('utf-8'))
        username = data['username']
        user = get_user(user_name=username)
        if data['register'] == 'yes':
            # register wallet to user
            wallet_ID = data['wallet_ID']
            if add_wallet(user, wallet_ID) is not None:
                response['register'] = 'success'
            else:
                response['register'] = 'failed'
        elif data['register'] == 'no':
            wallets_data = fetch_wallets_data(user=user)
            response['wallets_data'] = wallets_data
        return JsonResponse(response)